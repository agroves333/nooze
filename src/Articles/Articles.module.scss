.articles {
}

.grid {
    display: flex;
    margin-left: -10px; /* gutter size offset */
    width: auto;
}
.gridColumn {
    padding-left: 10px; /* gutter size */
    background-clip: padding-box;
}

.gridColumn > article { /* change div to reference your elements you put in <Masonry> */
   margin-bottom: 10px;
}