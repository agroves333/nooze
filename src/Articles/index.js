import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Masonry from 'react-masonry-css';

import Article from '../Article';
import styles from './Articles.module.css';

const breakpointColumns = {
  default: 3,
  900: 2,
  500: 1
};

class Articles extends Component {

  componentDidMount() {
  
  }
  
  renderArticles() {
    return this.props.data.map((article, key) => {
      return <Article key={`article-${key}`} data={article} />
    });
  }

  render() {
    return (
      <div className="articles">
        <Masonry
          breakpointCols={breakpointColumns}
          className={styles.grid}
          columnClassName={styles.gridColumn}>
          {this.renderArticles()}
        </Masonry>
      </div>
    );
  }
}

Articles.propTypes = {
  data: PropTypes.array,
};

Articles.defaultProps = {
  data: [],
};

export default Articles;
