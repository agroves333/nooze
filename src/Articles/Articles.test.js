import React from 'react';
import { render } from 'enzyme';
import Articles from './index';

const data = [
  {
    headline: {
      main: 'fake-headline',
    },
    snippet: 'fake-snippet',
    web_url: 'fake-url',
    multimedia: [{
      subtype: 'blog427',
      url: 'fake-url',
    }],
  },
  {
    headline: {
      main: 'fake-headline',
    },
    snippet: 'fake-snippet',
    web_url: 'fake-url',
    multimedia: [{
      subtype: 'blog427',
      url: 'fake-url',
    }],
  },
  {
    headline: {
      main: 'fake-headline',
    },
    snippet: 'fake-snippet',
    web_url: 'fake-url',
    multimedia: [{
      subtype: 'blog427',
      url: 'fake-url',
    }],
  },
];

it('renders 3 Articles', () => {
  const wrapper = render(<Articles data={data} />);
  const articles = wrapper.find('article');
  
  expect(articles).toHaveLength(3)
});

