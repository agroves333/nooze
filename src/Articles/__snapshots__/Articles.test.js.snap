// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Articles 1`] = `
<div
  className="articles"
>
  <Masonry
    breakpointCols={
      Object {
        "500": 1,
        "900": 2,
        "default": 3,
      }
    }
    className="grid"
    columnClassName="gridColumn"
  />
</div>
`;
